### ISSUE TITLE
ansible playbook not able to read aws profile credentials if export or passing parameter in yml 

### ISSUE THREAD

**Issue by gp-svc** [ansible/ansible#44647]
##### SUMMARY

ansible playbook not able to read aws profile credentials if export or passing parameter in yml 
i was working in ansible 2.3.0.0 it was fine. after updating ansible version aws related playbook not able to perform any action. 

> Issue identified - >even export profile and executing command or adding profile argument in 
playbook. ansible getting fails to perform actions. 
But if i pass access key and secret key then it is working fine.

> same case tested in 2.5 also getting same result

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

DescribeVpcs operation

##### ANSIBLE VERSION

```
ansible 2.6.3
  config file = /storage/home/gp/.ansible.cfg
  configured module search path = [u'/storage/home/gp/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

```

##### CONFIGURATION

RETRY_FILES_ENABLED(/storage/home/user/.ansible.cfg) = False

##### OS / ENVIRONMENT

"N/A"

##### STEPS TO REPRODUCE

```yaml


- name: Getting VPC_qa1 info
  hosts: localhost
  connection: local

  tasks:

  - ec2_vpc_net_facts:
      region: "us-west-2"
	  profile: "abc"
      filters:
        "tag:Environment": qa
    register: vpc_b

  
  - set_fact:
      my_vpcs:
            '{{ vpc_b.vpcs | map(attribute="id") | list }}'
      register: my_vpcs

  - name: Output content of VPC Facts
    debug: var=my_vpcs

  - name: Setting up vpc peering
    command: echo {{ item }}
    with_items:
      - "{{ my_vpcs }}"
    register: raw_result

```

##### EXPECTED RESULTS

TASK [Output content of VPC Facts] ********************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "peering_raw": {
        "changed": true, 
        "msg": "All items completed", 
        "results": [
            {
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": true, 
                "cmd": [
                    "echo", 
                    "vpc-vpc2id"
                ], 
                "delta": "0:00:00.012838", 
                "end": "2018-08-24 22:24:44.972364", 
                "invocation": {
                    "module_args": {
                        "_raw_params": "echo vpc-vpc2id", 
                        "_uses_shell": false, 
                        "chdir": null, 
                        "creates": null, 
                        "executable": null, 
                        "removes": null, 
                        "warn": true
                    }
                }, 
                "item": "vpc-vpc2id", 
                "rc": 0, 
                "start": "2018-08-24 22:24:44.959526", 
                "stderr": "", 
                "stderr_lines": [], 
                "stdout": "vpc-vpc2id", 
                "stdout_lines": [
                    "vpc-vpc2id"
                ]
            }, 
            {
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": true, 
                "cmd": [
                    "echo", 
                    "vpc-vpc1"
                ], 
                "delta": "0:00:00.007037", 
                "end": "2018-08-24 22:24:45.279928", 
                "invocation": {
                    "module_args": {
                        "_raw_params": "echo vpc-vpc1", 
                        "_uses_shell": false, 
                        "chdir": null, 
                        "creates": null, 
                        "executable": null, 
                        "removes": null, 
                        "warn": true
                    }
                }, 
                "item": "vpc-vpc1", 
                "rc": 0, 
                "start": "2018-08-24 22:24:45.272891", 
                "stderr": "", 
                "stderr_lines": [], 
                "stdout": "vpc-vpc1", 
                "stdout_lines": [
                    "vpc-vpc1"
                ]
            }
        ]
    }
}

##### ACTUAL RESULTS

```
TASK [ec2_vpc_net_facts] ******************************************************************************************************************************************************************************************
task path: /storage/home/gp/vpc_peer.yml:13
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: gp
<127.0.0.1> EXEC /bin/sh -c 'echo ~gp && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /storage/home/gp/.ansible/tmp/ansible-tmp-1535128723.16-101319043306264 `" && echo ansible-tmp-1535128723.16-101319043306264="` echo /storage/home/gp/.ansible/tmp/ansible-tmp-1535128723.16-101319043306264 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/cloud/amazon/ec2_vpc_net_facts.py
<127.0.0.1> PUT /storage/home/gp/.ansible/tmp/ansible-local-6126xnE88B/tmpIHot0S TO /storage/home/gp/.ansible/tmp/ansible-tmp-1535128723.16-101319043306264/ec2_vpc_net_facts.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /storage/home/gp/.ansible/tmp/ansible-tmp-1535128723.16-101319043306264/ /storage/home/gp/.ansible/tmp/ansible-tmp-1535128723.16-101319043306264/ec2_vpc_net_facts.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /storage/home/gp/.ansible/tmp/ansible-tmp-1535128723.16-101319043306264/ec2_vpc_net_facts.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /storage/home/gp/.ansible/tmp/ansible-tmp-1535128723.16-101319043306264/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_wd7gRy/ansible_module_ec2_vpc_net_facts.py", line 187, in describe_vpcs
    response = connection.describe_vpcs(VpcIds=vpc_ids, Filters=filters)
  File "/storage/home/gp/.local/lib/python2.7/site-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/storage/home/gp/.local/lib/python2.7/site-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
ClientError: An error occurred (AuthFailure) when calling the DescribeVpcs operation: AWS was not able to validate the provided access credentials

fatal: [localhost]: FAILED! => {
    "changed": false, 
    "error": {
        "code": "AuthFailure", 
        "message": "AWS was not able to validate the provided access credentials"
    }, 
    "invocation": {
        "module_args": {
            "aws_access_key": null, 
            "aws_secret_key": null, 
            "ec2_url": null, 
            "filters": {
                "tag:Environment": "qa"
            }, 
            "profile": "abc", 
            "region": "us-west-2", 
            "security_token": null, 
            "validate_certs": true, 
            "vpc_ids": []
        }
    }, 
    "msg": "Unable to describe VPCs []: An error occurred (AuthFailure) when calling the DescribeVpcs operation: AWS was not able to validate the provided access credentials", 
    "response_metadata": {
        "http_headers": {
            "date": "Fri, 24 Aug 2018 16:38:46 GMT", 
            "server": "AmazonEC2", 
            "transfer-encoding": "chunked"
        }, 
        "http_status_code": 401, 
        "request_id": "2c78891d-3300-46df-acb6-427419f3b2b3", 
        "retry_attempts": 0
    }
}
```

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ryansb** [comment]
In this case, the issue appears to be the indentation of the `profile:` argument. 

```
  - ec2_vpc_net_facts:
      region: "us-west-2"
	  profile: "abc"
      filters:
        "tag:Environment": qa
    register: vpc_b
```

Should be

```
  - ec2_vpc_net_facts:
      region: "us-west-2"
      profile: "abc"
      filters:
        "tag:Environment": qa
    register: vpc_b
```
Please try with fixed indentation and reopen if that does not resolve your issue.

**Comment by gp-svc** [comment]
apologies on this,  while copy past playbook content, filled with additional empty space. actually it have proper space. i will be sharing content again. 

  - ec2_vpc_net_facts:
      region: "us-west-2"
      profile: "abc"
      filters:
        "tag:Environment": qa1
    register: vpc_b

we are getting error as
task path: vpc_peer.yml:13
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: gp
<127.0.0.1> EXEC /bin/sh -c 'echo ~gp && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /storage/home/gp/.ansible/tmp/ansible-tmp-1535523529.78-53470824509002 `" && echo ansible-tmp-1535523529.78-53470824509002="` echo /storage/home/gp/.ansible/tmp/ansible-tmp-1535523529.78-53470824509002 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/cloud/amazon/ec2_vpc_net_facts.py
<127.0.0.1> PUT /storage/home/gp/.ansible/tmp/ansible-local-49414R7H3u/tmpkZO9cu TO /storage/home/gp/.ansible/tmp/ansible-tmp-1535523529.78-53470824509002/ec2_vpc_net_facts.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /storage/home/gp/.ansible/tmp/ansible-tmp-1535523529.78-53470824509002/ /storage/home/gp/.ansible/tmp/ansible-tmp-1535523529.78-53470824509002/ec2_vpc_net_facts.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /storage/home/gp/.ansible/tmp/ansible-tmp-1535523529.78-53470824509002/ec2_vpc_net_facts.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /storage/home/gp/.ansible/tmp/ansible-tmp-1535523529.78-53470824509002/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_Klh5dN/ansible_module_ec2_vpc_net_facts.py", line 187, in describe_vpcs
    response = connection.describe_vpcs(VpcIds=vpc_ids, Filters=filters)
  File "/storage/home/gp/.local/lib/python2.7/site-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/storage/home/gp/.local/lib/python2.7/site-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
ClientError: An error occurred (AuthFailure) when calling the DescribeVpcs operation: AWS was not able to validate the provided access credentials

fatal: [localhost]: FAILED! => {
    "changed": false, 
    "error": {
        "code": "AuthFailure", 
        "message": "AWS was not able to validate the provided access credentials"
    }, 
    "invocation": {
        "module_args": {
            "aws_access_key": null, 
            "aws_secret_key": null, 
            "ec2_url": null, 
            "filters": {
                "tag:Environment": "qa1"
            }, 
            "profile": "abc", 
            "region": "us-west-2", 
            "security_token": null, 
            "validate_certs": true, 
            "vpc_ids": []
        }
    }, 
    "msg": "Unable to describe VPCs []: An error occurred (AuthFailure) when calling the DescribeVpcs operation: AWS was not able to validate the provided access credentials", 
    "response_metadata": {
        "http_headers": {
            "date": "Wed, 29 Aug 2018 06:18:54 GMT", 
            "server": "AmazonEC2", 
            "transfer-encoding": "chunked"
        }, 
        "http_status_code": 401, 
        "request_id": "dd12b8db-f86b-468e-9158-9815c39beef5", 
        "retry_attempts": 0
    }
}

**Comment by kendaop** [comment]
I'm having the exact same issue as well, except I'm on Ansible version 2.5.0.

**Comment by jonTrent** [comment]
Ran into same issue when running an Ansible script to create a VPC.  (In my case it was via Ansible Tower.)  Error that I received was...

"Failed to describe VPCs: An error occurred (AuthFailure) when calling the DescribeVpcs operation"

Search led me here with no resolution.  For those also reaching this corner of the internet, I was able to resolve my problem by regenerating the AWS ACCESS KEY and SECRET KEY for the associated IAM account, which cleared up the issue.

I did notice that the AWS SECRET KEY of the offending credential included the "=" character, whereas the other two SECRET KEYS that I tried were essentially alphanumeric in addition to "/" and "+" characters.  Possibly the "=" character is not being handled properly by Ansible?!

Cheers.
